.github/workflows/main.yml

```diff
@@ -27,16 +27,26 @@ jobs:
       if: matrix.os == 'ubuntu-latest'
       run: |
          dotnet format --verify-no-changes --verbosity diagnostic
-    - name: Test
+    - name: Prepare coverage
+      if: matrix.os == 'ubuntu-latest'
```

Comment:
I think you need to prepare for all os in this case.
The idea I mentioned was to run just pure `dotnet test` on other matrix.os

Reply:
This can be optimized later, first I want to fix the coverage

Reply:
But now it's easier, less conditions, and all is separate